include/engine/routing_algorithms/shortest_path.hpp

```diff
@@ -2,22 +2,464 @@
 #define SHORTEST_PATH_HPP
 
 #include "engine/algorithm.hpp"
+#include "engine/datafacade.hpp"
 #include "engine/routing_algorithms/routing_base.hpp"
-#include "engine/search_engine_data.hpp"
+
 #include "util/typedefs.hpp"
 
+#include <boost/assert.hpp>
+#include <boost/optional.hpp>
+#include <memory>
+
 namespace osrm
 {
 namespace engine
 {
 namespace routing_algorithms
 {
 
+namespace
```

Comment:
I'm proposing we add this block to mainline OSRM so you can inject a header without having to modify OSRM code. I don't understand what you are trying to say.